lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.PlantLoop.rb

```diff
@@ -91,32 +91,70 @@ def plant_loop_apply_prm_number_of_cooling_towers(plant_loop)
   end
 
   # Splits the single chiller used for the initial sizing run
-  # into multiple separate chillers based on Appendix G.
-  #
+  # into multiple separate chillers based on Appendix G. Also applies
+  # EMS to stage chillers properly
   # @param plant_loop [OpenStudio::Model::PlantLoop] chilled water loop
-  # @param sizing_run_dir [String] sizing run directory
   # @return [Boolean] returns true if successful, false if not
-  def plant_loop_apply_prm_number_of_chillers(plant_loop, sizing_run_dir = nil)
+  def plant_loop_apply_prm_number_of_chillers(model, plant_loop)
     # Skip non-cooling plants & secondary cooling loop
     return true unless plant_loop.sizingPlant.loopType == 'Cooling'
     # If the loop is cooling but it is a secondary loop, then skip.
     return true if plant_loop.additionalProperties.hasFeature('is_secondary_loop')
 
-    # Determine the number and type of chillers
-    num_chillers = nil
-    chiller_cooling_type = nil
-    chiller_compressor_type = nil
-
     # Set the equipment to stage sequentially or uniformload if there is secondary loop
     if plant_loop.additionalProperties.hasFeature('is_primary_loop')
       plant_loop.setLoadDistributionScheme('UniformLoad')
     else
       plant_loop.setLoadDistributionScheme('SequentialLoad')
     end
 
+    # Get all existing chillers and pumps. Copy chiller properties needed when duplicating existing settings
+    chillers = []
+    pumps = []
+    default_cop = nil
+    condenser_water_loop = nil
+    dsgn_sup_wtr_temp_c = nil
+
+    plant_loop.supplyComponents.each do |sc|
+      if sc.to_ChillerElectricEIR.is_initialized
+        chiller = sc.to_ChillerElectricEIR.get
+
+        # Copy the last chillers COP, leaving chilled water temperature, and reference cooling tower. These will be the
+        # default for any extra chillers.
+        default_cop = chiller.referenceCOP
+        dsgn_sup_wtr_temp_c = chiller.referenceLeavingChilledWaterTemperature
+        condenser_water_loop = chiller.condenserWaterLoop
+        chillers << chiller
+
+      elsif sc.to_PumpConstantSpeed.is_initialized
+        pumps << sc.to_PumpConstantSpeed.get
+      elsif sc.to_PumpVariableSpeed.is_initialized
+        pumps << sc.to_PumpVariableSpeed.get
+      end
+    end
+
+    # Get existing plant loop pump. We'll copy this pumps parameters before removing it. Throw exception for multiple pumps on supply side
+    if pumps.size.zero?
+      OpenStudio.logFree(OpenStudio::Error, 'prm.log', "For #{plant_loop.name}, found #{pumps.size} pumps. A loop must have at least one pump.")
+      return false
+    elsif pumps.size > 1
+      OpenStudio.logFree(OpenStudio::Error, 'prm.log', "For #{plant_loop.name}, found #{pumps.size} pumps, cannot split up per performance rating method baseline requirements.")
```

Comment:
What does `split up per performance rating method` refers to here?

Reply:
I am unsure, that code block for `if pump.size.zero?` was in place before I refactored this and I moved it up. It looks like the last few updates to this code block were written by you and @weilixu :

https://github.com/NREL/openstudio-standards/blame/master/lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.PlantLoop.rb#L168

Reply:
I cannot recall this change I made...